Antidote/ChatView.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ChatView.swift
+//  Antidote
+//
+//  Created by Yoba on 03/01/2017.
+//  Copyright © 2017 Dmytro Vorobiov. All rights reserved.
+//
+
+import Cocoa
+import SnapKit
+
+// MARK: init methods and properties
+
+class ChatView: ListView {
+    var headerBar: NSView!
+    var bottomBar: NSView!
+    
+    struct Constants {
+        static let barHeight = 50
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    override init() {
+        super.init()
+        
+        createSubviews()
+        setupConstraints()
+    }
+}
+
+// MARK: subview handling
+
+fileprivate extension ChatView {
+    func createSubviews() {
+        headerBar = HeaderBarView()
+        addSubview(headerBar)
+        bottomBar = MessageBarView()
+        addSubview(bottomBar)
+    }
+    
+    func setupConstraints() {
+        headerBar.snp.makeConstraints {
+            $0.top.left.right.equalTo(self)
+            $0.height.equalTo(Constants.barHeight)
+            $0.bottom.equalTo(scrollView.snp.top)
+        }
+        
+        bottomBar.snp.makeConstraints {
+            $0.left.right.bottom.equalTo(self)
+            $0.top.equalTo(scrollView.snp.bottom)
+            $0.height.equalTo(Constants.barHeight)
+        }
+        
+        scrollView.snp.remakeConstraints {
+            $0.left.right.equalTo(self)
+            $0.top.equalTo(headerBar.snp.bottom)
+            $0.bottom.equalTo(bottomBar.snp.top)
```

Comment:
There is no need to duplicate constraints. You already set `headerBar.bottom <-> scrollView.top` on line 48, same for `bottomBar`.